### PR TITLE
Code Signing on Windows

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
This PR reenables code signing on Windows. 
Each Windows package built on CI should be now signed. 
Additionally, some refactorings were done around electron-builder config, so it is easier to use outside the build script and offers more configuration options.

[ci no changelog needed]

### Important Notes
Secrets are provided through GH page, certificate is deployed as part of CI runners configuration.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: app/ide-desktop/lib/client/electron-builder-config.ts [hunk 1]
```diff
@@ -39,23 +88,25 @@ const config: Configuration = {
     },
     win: {
         // We do not use compression as the build time is huge and file size saving is almost zero.
-        target: ['nsis'],
-        icon: `${icons}/icon.ico`,
+        target: args.targetOverride ?? 'nsis',
+        icon: `${args.iconsDist}/icon.ico`,
+        certificateFile: 'C:\\Users\\mwurb\\Downloads\\New Byte Order Sp. z o.o._cert_49517.p12',
```
Comment (MichaelMauderer) [inline app/ide-desktop/lib/client/electron-builder-config.ts]: Hard-coded path left over here.

File: app/ide-desktop/lib/client/electron-builder-config.ts [hunk 2]
```diff
@@ -101,6 +152,8 @@ const config: Configuration = {
     // TODO [mwu]: Temporarily disabled, signing should be revised.
```
Comment (MichaelMauderer) [inline app/ide-desktop/lib/client/electron-builder-config.ts:150]: Is this TODO affected by this PR?

**Review by MichaelMauderer — CHANGES_REQUESTED** [review]

File: app/ide-desktop/lib/client/electron-builder-config.ts (hunk 1 above)
Comment (mwu-tow) [reply app/ide-desktop/lib/client/electron-builder-config.ts]: Fixed.

File: app/ide-desktop/lib/client/electron-builder-config.ts (hunk 2 above)
Comment (mwu-tow) [reply app/ide-desktop/lib/client/electron-builder-config.ts:150]: No, the TODO is about macOS code signing, this PR addresses the Windows code signing.

**Pull request merged**